bigtable/cmd/emulator/cbtemulator.go

```diff
@@ -21,30 +21,71 @@ import (
 	"flag"
 	"fmt"
 	"log"
+	"math/rand"
+	"strings"
+	"time"
 
 	"cloud.google.com/go/bigtable/bttest"
 	"google.golang.org/grpc"
 )
 
 var (
-	host = flag.String("host", "localhost", "the address to bind to on the local machine")
-	port = flag.Int("port", 9000, "the port number to bind to on the local machine")
+	host                   = flag.String("host", "localhost", "the address to bind to on the local machine")
+	port                   = flag.Int("port", 9000, "the port number to bind to on the local machine")
+	introduce_grpc_latency = flag.Bool("introduce_grpc_latency", false, "Introduce gRPC latency for testing")
+	grpc_read_p50          = flag.Duration("grpc_p50_read", time.Duration(0), "Target gRPC latency for ReadRows method at p50. Valid if introduce_grpc_latency is set.")
+	grpc_read_p99          = flag.Duration("grpc_p99_read", time.Duration(0), "Target gRPC latency for ReadRows method at p99. Valid if introduce_grpc_latency is set.")
+	grpc_write_p50         = flag.Duration("grpc_p50_write", time.Duration(0), "Target gRPC latency for MutateRows method at p50. Valid if introduce_grpc_latency is set.")
+	grpc_write_p99         = flag.Duration("grpc_p99_write", time.Duration(0), "Target gRPC latency for MutateRows method at p99. Valid if introduce_grpc_latency is set.")
```

Comment:
To be complete there should be knobs for singular MutateRow, CheckAndMutate and ReadModifyWrite. In terms of ui, I think it would be nicer to do something a bit more generic:

`--inject-latency="MutateRow:p50:10ms" --inject-latency="MutateRow:p99:100ms" `

Reply:
Thanks for feedback - updated code to follow suggestions.  Please lmk if this approach looks reasonable. Will add tests if so